lib/eslint.js

```diff
@@ -994,21 +994,32 @@ module.exports = (function() {
      */
     api.markVariableAsUsed = function(name) {
         var scope = this.getScope(),
-            variables,
-            i,
-            len;
-
-        do {
-            variables = scope.variables;
-            for (i = 0, len = variables.length; i < len; i++) {
-                if (variables[i].name === name) {
-                    variables[i].eslintUsed = true;
-                    return true;
+            found = false;
+
```

Comment:
There's a much more appropriate change to make here. This could cause the wrong variable to be marked because it searches global and then node scope, which is the opposite of how variable resolution works.

Instead, this can be added to the original code just before the loop:

```js
if (scope.type === "global" && currentConfig.ecmaFeatures.global return) {
    scope = scope.childScopes[0];
}
```